pkg/security/security_profile/secprofs.go

```diff
@@ -439,6 +439,10 @@ func (m *Manager) onWorkloadSelectorResolvedEvent(workload *tags.Workload) {
 				seclog.Warnf("couldn't load profile from local storage: %v", err)
 				return
 			} else if ok {
+
+				// apply the eviction mechanism right away
+				p.ActivityTree.EvictUnusedNodes(time.Now().Add(-m.config.RuntimeSecurity.SecurityProfileNodeEvictionTimeout), m.GetNodesInProcessCache())
```

Comment:
I wonder if we should do the same for when we got the profile from the cache (like at line 419). WDYT ?